src/services/firebase.ts

```diff
@@ -107,9 +115,95 @@ export namespace ourFirebase {
     const matchIds = Object.keys(matchMemberships);
     // TODO: get all matches in one call to firebase, then later call dispatch.
     // Make sure we listen to match changes only once.
-    // '/gamePortal/matches'
-    // dispatch(updateMatchList);
-    getRef('/gamePortal/matches' + matchIds); // TODO
+    // 'gamePortal/matches'
+    // store.dispatch(updateMatchList);
+    let tempMatchIds: Promise<any>[] = [];
+    for (let matchId of matchIds) {
+      // getMatchDetail(matchId).then(() => {tempMatchIds.push(getMatchDetail(matchId))});
+      const match = getMatchDetail(matchId);
+
+      match.then(data => {
```

Comment:
Remove this then.

Reply:
But how can I "filter the failed promise" as you said if I don't us then here?  I add a status field in promise return result to check if it returns null or not. 

Can I get the value that the promise returns if I don't use a then?